### ISSUE TITLE
bug(HOCs): Static methods are not copied over

### ISSUE THREAD

**Issue by jeloagnasin** [prescottprue/react-redux-firebase#732]
**Do you want to request a *feature* or report a *bug*?**  

Bug

**What is the current behavior?**

Static methods are not copied over if I use `firebaseConnect`.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**

See: https://github.com/jeloagnasin/rrf-testing/blob/master/App.js for minimal demo.

In the demo, without the `firebaseConnect()`, the title of the screen will show up in the navigation header. But if I wrap my component in `firebaseConnect()`, the title is not displayed. Essentially, `navigationOptions` is not called.

```javascript
const Stack = createAppContainer(createStackNavigator({
  // Testing: firebaseConnect()(Testing),
  Testing,
}));
```

**What is the expected behavior?**

Static methods are copied over like in v2.3.0.

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**

* react-redux-firebase@^3.0.0-alpha.13 (worked in 2.3.0)
* react-redux@^6.0.0

If you need more info, let me know. Thanks!

**Comment by jeloagnasin** [comment]
Workarounds for this issue are:

```javascript
const WrappedTesting = firebaseConnect()(Testing);
WrappedTesting.navigationOptions = Testing.navigationOptions;

const Stack = createAppContainer(createStackNavigator({
  Testing: WrappedTesting,
}));
```

or

```javascript
import hoistStatics from 'hoist-non-react-statics';

const WrappedTesting = firebaseConnect()(Testing);
const Stack = createAppContainer(createStackNavigator({
  Testing: hoistStatics(WrappedTesting, Testing),
}));
```

**Comment by jeloagnasin** [comment]
Since #733 is a temporary fix, check out #734 for a permanent solution. illuminst is refactoring `firebaseConnect` and `firestoreConnect` to use hooks.

**Comment by prescottprue** [comment]
In order to keep backwards compatibility with react versions without hooks for `firebaseConnect` and `firestoreConnect` for v3.0.0 I switched them back to classes, but included similar changes to what you did in #733. So it should be good now - let me know if things aren't functioning as expected and we can reopen
